persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java

```diff
@@ -1438,7 +1440,7 @@ protected BuiltSelect selectQuery() {
         valueCodec,
         executor,
         builder,
-        new HashSet<>(selectedColumns),
+        allSelected,
```

Comment:
Does this change semantics? `allSelected` seems to be modified after it is created from `selectedColumns` :thinking:

Reply:
It does, @pcmanus confirmed that it was a legitimate bug.